resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -182,10 +185,25 @@ private[spark] class KubernetesClusterSchedulerBackend(
       } else if (totalExpectedExecutors.get() <= runningExecutorPods.size) {
         logDebug("Maximum allowed executor limit reached. Not scaling up further.")
       } else {
+        val executorPodsWithIPs = EXECUTOR_PODS_BY_IPS_LOCK.synchronized {
+            executorPodsByIPs.values.toList
```

Comment:
Yes, it does. The output is an immutable list while the map is a mutable hashmap, so the toList copies the values to make an immutable list. I'll add a comment.

```
scala> val a = scala.collection.mutable.HashMap("a" -> 1, "b" -> 2)
a: scala.collection.mutable.HashMap[String,Int] = Map(b -> 2, a -> 1)

scala> val b = a.values.toList
b: List[Int] = List(2, 1)

scala> a -= "b"
res1: a.type = Map(a -> 1)

scala> a.values
res2: Iterable[Int] = HashMap(1)

scala> b
res3: List[Int] = List(2, 1)

scala> a -= "a"
res4: a.type = Map()

scala> a.values
res5: Iterable[Int] = HashMap()

scala> b
res6: List[Int] = List(2, 1)
```